modules/solid_mechanics/include/materials/NewtonIterationSingleVariable.h

```diff
@@ -0,0 +1,214 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef NEWTONITERATIONSINGLEVARIABLE_H
+#define NEWTONITERATIONSINGLEVARIABLE_H
+
+#include "InputParameters.h"
+#include "ConsoleStream.h"
+
+/**
+ * Base class that provides capability for Newton return mapping
+ * iterations on a single variable
+ */
+
+class NewtonIterationSingleVariable;
+
+template <>
+InputParameters validParams<NewtonIterationSingleVariable>();
+
+class NewtonIterationSingleVariable
+{
+public:
+  NewtonIterationSingleVariable(const InputParameters & parameters);
+  virtual ~NewtonIterationSingleVariable() {}
+
+protected:
+  /**
+   * Perform the return mapping iterations
+   * @param effective_trial_stress Effective trial stress
+   * @param scalar                 Inelastic strain increment magnitude being solved for
+   * @param console                Console output
```

Comment:
I believe this is how we're supposed to document the code for Doxygen.  I've personally never been very convinced of the benefits of Doxygen, but I'm pretty sure that's the coding standard.